cpp/src/arrow/compute/kernels/scalar_string_test.cc

```diff
@@ -56,12 +56,14 @@ TYPED_TEST(TestStringKernels, AsciiLength) {
                    "[3, null, 0, 1]");
 }
 
-TYPED_TEST(TestStringKernels, DISABLED_AsciiUpper) {
+TYPED_TEST(TestStringKernels, AsciiUpper) {
+  this->CheckUnary("ascii_upper", "[]", this->string_type(), "[]");
```

Comment:
Does this implicitly test sliced arrays?

Reply:
It does now -- @bkietz added this in ARROW-971